aten/src/ATen/benchmarks/stateful_conv1d.cpp

```diff
@@ -0,0 +1,85 @@
+  const size_t input_channels = static_cast<size_t>(state.range(0));
+  const size_t output_channels = static_cast<size_t>(state.range(1));
+  const size_t kernel = static_cast<size_t>(state.range(2));
+  const size_t batch_size = static_cast<size_t>(state.range(3));
+  const size_t width = static_cast<size_t>(state.range(4));
+  const bool optimized = static_cast<bool>(state.range(5));
+
+  torch::jit::Module m("m");
+  m.register_parameter("weight_1", torch::rand({output_channels, input_channels, kernel}), false);
+  m.register_parameter("bias_1", torch::rand({output_channels}), false);
+  m.register_parameter("weight_2", torch::rand({output_channels, output_channels, kernel}), false);
+  m.register_parameter("bias_2", torch::rand({output_channels}), false);
+  m.register_parameter("weight_3", torch::rand({output_channels, output_channels, kernel}), false);
+  m.register_parameter("bias_3", torch::rand({output_channels}), false);
+  m.register_parameter("weight_4", torch::rand({output_channels, output_channels, kernel}), false);
+  m.register_parameter("bias_4", torch::rand({output_channels}), false);
+
+  m.define(R"(
+    def forward(self, x):
+      x = torch.conv1d(x, self.weight_1, self.bias_1, 1, 0, 1, 1)
+      x = torch.conv1d(x, self.weight_2, self.bias_2, 1, 0, 1, 1)
+      x = torch.conv1d(x, self.weight_3, self.bias_3, 1, 0, 1, 1)
+      x = torch.conv1d(x, self.weight_4, self.bias_4, 1, 0, 1, 1)
+      return x
+  )");
+
+  std::vector<std::vector<torch::jit::IValue>> inputs;
+  for (int i = 0; i < 10; ++i) {
+    std::vector<torch::jit::IValue> input;
+    input.push_back(torch::rand({batch_size, input_channels, width}));
+    inputs.push_back(input);
+  }
+
+  auto m_cloned = m.clone();
+  torch::jit::transformConv1dToConv2d(m_cloned);
+  auto m_optimized = torch::jit::optimizeForMobile(m_cloned);
+  torch::jit::IValue output;
+
+  if (!optimized) {
+    for (auto _ : state) {
+      for (const auto& input : inputs) {
+        output = m.forward(input);
+      }
+    }
+  } else {
+    for (auto _ : state) {
+      for (const auto& input : inputs) {
+        output = m_optimized.forward(input);
+      }
+    }
+  }
+}
+
+static void GenerateSizes(benchmark::internal::Benchmark* b) {
+  b->ArgNames({"Input Channels",
+               "Output Channels",
+               "Kernel",
+               "Batch Size",
+               "Width",
+               "Optimized"});
+
+  for (size_t input_channels = 32; input_channels < 256; input_channels *= 2) {
+    for (size_t output_channels = 32; output_channels < 256; output_channels *= 2) {
+      for (size_t kernel = 3; kernel < 8; ++kernel) {
+        for (size_t batch_size = 1; batch_size < 5; ++batch_size) {
+          for (size_t width = 32; width < 256; width *= 2) {
+            b->Args({input_channels, output_channels, kernel, batch_size, width, true});
+            b->Args({input_channels, output_channels, kernel, batch_size, width, false});
```

Comment:
I think you want two benchmark, no? Otherwise how are you distinguish the reported results between optimized vs unoptimized? Looking at the <state> values reported by benchmark? I think that is fine as well.

Reply:
So currently if we look at two consecutive lines, we can see the results for the two cases: optimized (Optimzed:1) and unoptimized(Optimzed:0). I think that way we can easily compare the performance for a given set of hyperparameter. Let me know if that's ok.

Reply:
Sure, thats ok.